integration/joinopenssh_test.go

```diff
@@ -0,0 +1,155 @@
+	rcConf.Proxy.DisableWebInterface = true
+	rcConf.Version = "v3"
+	rcConf.Auth.StaticTokens, err = types.NewStaticTokens(types.StaticTokensSpecV2{
+		StaticTokens: []types.ProvisionTokenV1{
+			{
+				Roles: []types.SystemRole{types.RoleNode},
+				Token: "token",
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	err = rc.CreateEx(t, nil, rcConf)
+	require.NoError(t, err)
+	err = rc.Start()
+	require.NoError(t, err)
+	defer rc.StopAll()
+
+	ctx := context.Background()
+
+	opensshConfigPath := filepath.Join(testDir, "sshd_config")
+	f, err := os.Create(opensshConfigPath)
+	require.NoError(t, err)
+	require.NoError(t, f.Close())
+	restartPath := filepath.Join(testDir, "restarted")
+	teleportOpenSSHCfgPath := filepath.Join(testDir, "teleport_openssh")
+
+	openSSHCfg := servicecfg.MakeDefaultConfig()
+
+	openSSHCfg.OpenSSH.Enabled = true
+	err = config.ConfigureOpenSSH(&config.CommandLineFlags{
+		DataDir:           teleportOpenSSHCfgPath,
+		ProxyServer:       rc.Web,
+		AuthToken:         "token",
+		JoinMethod:        string(types.JoinMethodToken),
+		OpenSSHConfigPath: opensshConfigPath,
+		RestartOpenSSH:    true,
+		RestartCommand:    fmt.Sprintf("touch %q", restartPath),
+		CheckCommand:      "echo okay",
+		Labels:            "hello=true",
+		Address:           "127.0.0.1:22",
+		InsecureMode:      true,
+		Debug:             true,
+	}, openSSHCfg)
+	require.NoError(t, err)
+
+	err = service.Run(ctx, *openSSHCfg, nil)
+	require.NoError(t, err)
+
+	client := rc.GetSiteAPI(rc.Secrets.SiteName)
+
+	sshdConf, err := os.ReadFile(opensshConfigPath)
+	require.NoError(t, err)
+	require.Contains(t, string(sshdConf), fmt.Sprintf("Include %s", filepath.Join(teleportOpenSSHCfgPath, "sshd.conf")))
+
+	// check a node with the flags specified exists
+	require.Eventually(t, findNodeWithLabel(t, ctx, client, "hello"), time.Second*2, time.Millisecond*50)
+	// check the ssh command was in fact called
+	require.FileExists(t, restartPath)
+
+	keysDir := filepath.Join(teleportOpenSSHCfgPath, "openssh")
+	// check all the appropriate key files were made
+	require.DirExists(t, keysDir)
+	cabytes, err := os.ReadFile(filepath.Join(keysDir, openssh.TeleportOpenSSHCA))
+	require.NoError(t, err)
+	require.FileExists(t, filepath.Join(keysDir, openssh.TeleportCert))
+	require.FileExists(t, filepath.Join(keysDir, openssh.TeleportKey))
+
+	allOpenSSHCAs := getOpenSSHCAs(t, ctx, client)
+	require.ElementsMatch(t, bytes.Split(bytes.TrimSpace(cabytes), []byte("\n")), allOpenSSHCAs)
+}
+
+func getOpenSSHCAs(t *testing.T, ctx context.Context, cl auth.ClientI) [][]byte {
+	t.Helper()
+	cas, err := cl.GetCertAuthorities(ctx, types.OpenSSHCA, false)
+	require.NoError(t, err)
+	var caBytes [][]byte
+	for _, ca := range cas {
+		for _, key := range ca.GetTrustedSSHKeyPairs() {
+			caBytes = append(caBytes, bytes.TrimSpace(key.PublicKey))
+		}
+	}
+	return caBytes
+}
+
+func findNodeWithLabel(t *testing.T, ctx context.Context, cl auth.ClientI, key string) func() bool {
```

Comment:
Performance doesn't matter since this cluster has no resources, but you could remove some code by using `cl.ListResources` here instead.

(`ListResources` allows you to search by label, so you don't have to request all nodes and then check their labels)